pkg/plugins/plugins.go

```diff
@@ -53,65 +56,81 @@ func Setup(client *Client, plugins map[string]Descriptor, devPlugin *DevPlugin)
 		}
 	}
 
-	if devPlugin != nil {
-		err := checkDevPluginConfiguration(devPlugin)
-		if err != nil {
-			return fmt.Errorf("invalid configuration: %w", err)
-		}
-	}
-
 	return nil
 }
 
-func checkDevPluginConfiguration(plugin *DevPlugin) error {
-	if plugin == nil {
+// SetupLocalPlugins setup local plugins environment.
+func SetupLocalPlugins(plugins map[string]LocalDescriptor) error {
+	if plugins == nil {
 		return nil
 	}
 
-	if plugin.GoPath == "" {
-		return errors.New("missing Go Path (prefer a dedicated Go Path)")
-	}
+	uniq := make(map[string]struct{})
+
+	var errs *multierror.Error
+	for pAlias, descriptor := range plugins {
+		if descriptor.ModuleName == "" {
+			errs = multierror.Append(errs, fmt.Errorf("%s: plugin name is missing", pAlias))
+		}
+
+		if strings.HasPrefix(descriptor.ModuleName, "/") || strings.HasSuffix(descriptor.ModuleName, "/") {
+			errs = multierror.Append(errs, fmt.Errorf("%s: plugin name should not start or end with a /", pAlias))
+			continue
+		}
+
+		if _, ok := uniq[descriptor.ModuleName]; ok {
+			errs = multierror.Append(errs, fmt.Errorf("only one version of a plugin is allowed, there is a duplicate of %s", descriptor.ModuleName))
+			continue
+		}
 
-	if plugin.ModuleName == "" {
-		return errors.New("missing module name")
+		uniq[descriptor.ModuleName] = struct{}{}
+
+		err := checkLocalPluginManifest(descriptor)
+		errs = multierror.Append(errs, err)
```

Comment:
Maybe we can inline to avoid confusion with simple error management.
```suggestion
		errs = multierror.Append(errs, checkLocalPluginManifest(descriptor))
```